File: samples/csharp/src/Trees.cs
using System;
using System.Collections.Generic;

internal class Trees
{
    public static void RunMain(string[] args)
    {
        new Trees().Start();
    }

    public void Start()
    {
        // // nodes of a tree
        Node<string> root = new Node<string>("0");
        Node<string> node1 = new Node<string>("1");
        Node<string> node2 = new Node<string>("2");
        Node<string> node3 = new Node<string>("3");
        Node<string> node4 = new Node<string>("4");
        Node<string> node5 = new Node<string>("5");
        Node<string> node6 = new Node<string>("6");
        root.Add(node1);
        node1.Add(node2);
        node1.Add(node3);
        root.Add(node4);
        node4.Add(node5);
        node4.Add(node6);

        // // wrap to a tree
        Tree<string> tree = new Tree<string>(root);
        visitTree(tree.Root, tree.Root);

        Console.WriteLine("done");
    }

    private static void visitTree(Node<string> node, Node<string> root)
    {
        if (node == null)
        {
            return;
        }
        Console.WriteLine(node.Value);
        foreach (Node<string> child in node.children)
        {
            visitTree(child, root);
        }
    }
}

class Tree<T>
{

    public Node<T> Root { get; private set; }

    public Tree(Node<T> root)
    {
        this.Root = root;
    }
}

class Node<T>
{

    public T Value { get; private set; }
    public List<Node<T>> children = new List<Node<T>>();

    public Node(T value)
    {
        this.Value = value;
    }

    public void Add(Node<T> value)
    {
        children.Add(value);
    }
}

File: samples/python/src/Trees.py
from __future__ import annotations
from typing import TypeVar, Generic
from collections import deque

T = TypeVar("T")


class Trees:

    @staticmethod
    def run_main():
        Trees().start()

    def start(self):
        # nodes of a tree
        root: Node[str] = Node[str]("0")
        node1: Node[str] = Node[str]("1")
        node2: Node[str] = Node[str]("2")
        node3: Node[str] = Node[str]("3")
        node4: Node[str] = Node[str]("4")
        node5: Node[str] = Node[str]("5")
        node6: Node[str] = Node[str]("6")

        root.add(node1)
        node1.add(node2)
        node1.add(node3)
        root.add(node4)
        node4.add(node5)
        node4.add(node6)

        # wrap to a tree
        tree: Tree[str] = Tree[str](root)
        Trees.visit_tree(tree.get_root(), tree.get_root())

        print("done")

    @staticmethod
    def visit_tree(node: Node[str], root: Node[str]):
        if node is None:
            return
        print(node.value)
        for child in node.children:
            Trees.visit_tree(child, root)


class Tree(Generic[T]):

    def __init__(self, root: Node[T]):
        self.root = root

    def get_root(self) -> Node[T]:
        return self.root


class Node(Generic[T]):
    def __init__(self, value: T):
        self.children: list[Node[T]] = []
        self.value: T = value

    def add(self, value: Node[T]):
        self.children.append(value)

    def __str__(self) -> str:
        return str(self.value)


Trees.run_main()
